src/actions/tests/breakpoints.js

```diff
@@ -38,13 +48,18 @@ describe("breakpoints", () => {
 
   describe("adding a breakpoint to an invalid location", async () => {
     it("adds only one breakpoint with a corrected location", async () => {
-      const invalidLocation = { sourceId: "a", line: 5 };
+      const invalidLocation = {
+        sourceId: "a",
+        line: 5,
+        sourceUrl: "http://localhost:8000/examples/a"
+      };
       const {
         correctedThreadClient,
         correctedLocation
       } = simulateCorrectThreadClient(2, invalidLocation);
       const { dispatch, getState } = createStore(correctedThreadClient);
 
+      await dispatch(actions.newSource(makeSource("a")));
```

Comment:
we're now setting sources because you can't add a breakpoint without a corresponding source. This is important because we want source level information in a bp like sourceUrl and sourceId